runtime/util_core/j9shchelp.c

```diff
@@ -129,6 +129,36 @@ getGenerationFromName(const char* cacheNameWithVGen)
 	}
 }
 
+/**
+ * Return the value of the layer number from the cache filename
+ *
+ * @param [in] cacheNameWithVGen  the cache name with layer number
+ *
+ * @return J9SH_LAYER_NUM_UNSET(-1) if an error occurred or there is no layer number in file name.
+ * 		   Otherwise return the layer number
+ */
+I_8
+getLayerFromName(const char* cacheNameWithVGen)
+{
+	char* cursor = (char*)cacheNameWithVGen;
+	IDATA nameLen = strlen(cacheNameWithVGen);
+	I_8 layerNumber = J9SH_LAYER_NUM_UNSET;
+	UDATA temp0 = 0;
+
+	cursor += (nameLen - 3);
```

Comment:
There should be a check that `nameLen > 3`.

Reply:
Check added.